src/Core/hco_restart_mod.F90

```diff
@@ -330,9 +330,9 @@ SUBROUTINE HCO_RestartGet_3D( HcoState, Name,   Arr3D, &
         RETURN
     ENDIF
 
-    ! If field is all zero assume it to be not filled
+    ! If field is all negatives assume it to be not filled
     IF ( FLD ) THEN
```

Comment:
Hi @christophkeller, we ran into an issue in the GCHP benchmark which Bob and I traced to this code. It went under the radar before since I was unaware we used hco_restart_mod in GCHP. Is this change supposed to correspond to a MAPL version update? As is, if a restart variable is not in the GCHP restart file it will be bootstrapped to zero. Is it now set to negative values?

Reply:
I am editing this to include all zeros as well as all negatives in the criteria for restart field not filled. Let me know if that's an issue for GEOS.

Reply:
Checking for all zeros is dangerous because some restart files are zero in most places except for a few. Since this check is performed on each CPU individually, it can then quite easily happen that the code accidentally assumes a restart field to be undefined over some CPUs (but not others). The check for all negatives is safer in that regard (albeit also not ideal). Setting it back to all zero might work for GCHP but will be problematic for GEOS.

One option for bootstrapping a restart field is to create a dummy field in your gchp_internal_rst with the variable name of interest and with all values set to -999. This is how I usually bootstrap fields for GEOS.

Reply:
Okay, sounds good. I'll to put the new code in MODEL_GEOS ifdefs as a quick fix for 13.4.0 release and then work on a better solution for 14.0.